backend/src/main/java/eventlistener/service/event/EventService.java

```diff
@@ -82,15 +80,15 @@ private void clearAllEventLinks(NotificationUser user){
         eventRepo.saveAll(userEvents);
     }
 
-    private boolean userExistInEvent(long id, Event toCheck){
-        return toCheck.getNotificationUser().contains(id);
+    private boolean userExistInEvent(NotificationUser user, Event toCheck){
+        return toCheck.getNotificationUser().contains(user);
```

Comment:
is the notificationUser attribute of the Event a collection? If so it should be called notificationUsers